outputs.tf

```diff
@@ -1,23 +1,28 @@
 output "ngw_id" {
   value       = "${join("", aws_nat_gateway.default.*.id)}"
-  description = "IDs of NAT Gateways"
+  description = "NAT Gateway ID"
 }
 
 output "ngw_private_ip" {
   value       = "${join("", aws_nat_gateway.default.*.private_ip)}"
-  description = "The private IP addresses of the NAT Gateways"
+  description = "Private IP address of the NAT Gateway"
 }
 
 output "ngw_public_ip" {
   value       = "${join("", aws_nat_gateway.default.*.public_ip)}"
-  description = "The public IP addresses of the NAT Gateways"
+  description = "Public IP address of the NAT Gateway"
 }
 
 output "subnet_ids" {
   value       = ["${coalescelist(aws_subnet.private.*.id, aws_subnet.public.*.id)}"]
-  description = "IDs of private subnets"
+  description = "Subnet IDs"
 }
 
 output "route_table_ids" {
-  value = ["${coalescelist(aws_route_table.public.*.id, aws_route_table.private.*.id)}"]
+  value       = ["${coalescelist(aws_route_table.public.*.id, aws_route_table.private.*.id)}"]
+  description = "Route Table IDs"
+}
+
+output "subnet_names_subnet_ids" {
+  value = "${zipmap(var.names, matchkeys(coalescelist(aws_subnet.private.*.id, aws_subnet.public.*.id), coalescelist(aws_subnet.private.*.tags.Named, aws_subnet.public.*.tags.Named), var.names))}"
```

Comment:
Are we sure, that `matchkeys` returns ordered list according to `var.names`?

Certain module instance will return only subnets of certain `type` (either `public` or `private`).
The following variant shorter, and provides the same result.
```
${zipmap(var.names, coalescelist(aws_subnet.private.*.id, aws_subnet.public.*.id))}
```